### ISSUE TITLE
Allow using the latest version with older versions of NGINX Plus

### ISSUE THREAD

**Issue by lucacome** [nginxinc/nginx-prometheus-exporter#629]
I want to be able to use the exporter with older versions of NGINX Plus and not have the latest version of the API hardcoded.

Depends on https://github.com/nginxinc/nginx-plus-go-client/issues/222

**Comment by alitoufighi** [comment]
My assumption is we won't be able to fix this with the given solution for https://github.com/nginxinc/nginx-plus-go-client/issues/222 as it needs an already connected client.

**Comment by lucacome** [comment]
@alitoufighi yes we need a connection to get the API versions from NGINX. One of the options that I was thinking about was to check the version every time we collect the metrics, but that might be a little bit expensive.

I'm open to suggestions 🙂

**Comment by alitoufighi** [comment]
Hmm maybe beginning the connection in a minimal way (is finding the lowest compatible api version any easier? i.e. `1`?) and then promoting it to a higher version? 

Also I don't think the call to get API versions is over a specific API version, so it could be doable to communicate over to find the max compatible API version during initialization as well, but it might not be the cleanest way (might actually give this a shot)

**Comment by mpstefan** [comment]
Blocked until nginx-plus-go-client 1.3.0 is release. When it is, @alitoufighi would you be open to creating a PR for this?

**Comment by sindhushiv** [comment]
nginx-plus-go-client is available so @alitoufighi you can work on this if you want.
